src/ExtConnectionInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace React\Socket;
+
+/**
+ * This is an extended connection interface,
+ * which allows to expose the underlying stream
+ * in a non-BC way. This interface will be removed
+ * in the next major release and merge into `ConnectionInterface`.
+ *
+ * However it is not exposed for arbitrary reasons.
+ * Most notably it is exposed to enable TLS.
+ */
+interface ExtConnectionInterface extends ConnectionInterface
+{
+    /**
+     * Returns the underlying stream.
+     * @return resource
+     */
+    public function getStream();
+
+    /**
+     * Sets the internal flag to specify whether
+     * TLS was enabled or not.
+     *
+     * This is used to return `tls://` or `tcp://` as scheme.
+     * @param bool $flag
+     * @return void
+     */
+    public function setTLSEnabledFlag($flag);
```

Comment:
Not sure how I feel about this interface. Have you considered splitting this into two different interfaces? `ExtConnectionInterface` feels rather general and we might need a `ExtExtConnectionInterface` later on if we'd decide to expand the interface again. /cc @clue

Reply:
We could make two interfaces yes, and at this point we should also split `Connection` into two classes - one for TCP and one for UDS.

Reply:
@WyriHaximus `RealExtConnectionInterface` 😜

Reply:
@kelunik `DontDrinkTheCoolAidRealExtConnectionInterface` :sunglasses: 
@CharlotteDunois Didn't think that far ahead tbh. The thing that mainly bugs me is that I would hate to to see `ExtExtExt` interfaces down the line in a while

Reply:
@WyriHaximus Well I did. We can certainly split this off and give better names. However we do need to split the implementation class too when we go that route. I'm willed to create a followup pr for this.

Reply:
@CharlotteDunois I'd be up for that, you as well @clue and @jsor ?